### PR TITLE
chore: fix precommit hooks

### PR DESCRIPTION
**Pull request by timsuchanek**
- Enable pre-commit hooks
- Add linting to Buildkite CI

### CODE REVIEW

File: src/packages/client/src/__tests__/integration/happy/disconnect-finally/test.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { getTestClient } from '../../../../utils/getTestClient'
 
+/* eslint-disable */
```
Comment (Jolg42) [inline src/packages/client/src/__tests__/integration/happy/disconnect-finally/test.ts:3]: Why disabling?

File: src/packages/client/src/generation/utils.ts [hunk 2]
```diff
@@ -133,7 +135,7 @@ export function getDefaultArgName(
 export function getOperation(action: DMMF.ModelAction): 'query' | 'mutation' {
   if (
     action === DMMF.ModelAction.findMany ||
-    action === DMMF.ModelAction.findUnique || 'findOne'
+    action === DMMF.ModelAction.findUnique
```
Comment (Jolg42) [inline src/packages/client/src/generation/utils.ts:138]: I guess that's a fix unrelated to linting

File: src/packages/debug/package.json [hunk 3]
```diff
@@ -36,19 +36,5 @@
   ],
   "dependencies": {
     "debug": "^4.1.1"
-  },
-  "husky": {
-    "hooks": {
-      "pre-commit": "lint-staged"
-    }
-  },
-  "lint-staged": {
```
Comment (Jolg42) [inline src/packages/debug/package.json:45]: If this was removed then we don't need the dev dependencies associated?

File: src/packages/get-platform/pnpm-lock.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+dependencies:
```
Comment (Jolg42) [inline src/packages/get-platform/pnpm-lock.yaml:1]: I think you can delete this lockfile? or does every new package comes with its own now?

**Review by Jolg42 — APPROVED** [review]
Looks good (see comments).

It's massive :D

**Comment by Jolg42** [comment]
Looks like there is an error in CI
```
Error: src/runtime/query.ts(880,9): error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'ArgError'.
  Type '{}' is missing the following properties from type 'ArgError': path, error
```

File: src/packages/debug/package.json (hunk 3 above)
Comment (Jolg42) [reply src/packages/debug/package.json:45]: this needs to be checked in all pkg

File: src/packages/get-platform/pnpm-lock.yaml (hunk 4 above)
Comment (Jolg42) [reply src/packages/get-platform/pnpm-lock.yaml:1]: this needs to be checked in all pkg

File: src/packages/client/src/generation/utils.ts (hunk 2 above)
Comment (Jolg42) [reply src/packages/client/src/generation/utils.ts:138]: It's a bit weird that "'findOne'" was only removed here?

File: src/packages/debug/package.json (hunk 3 above)
Comment (timsuchanek) [reply src/packages/debug/package.json:45]: This will come back, just wip

File: src/packages/client/src/generation/utils.ts (hunk 2 above)
Comment (timsuchanek) [reply src/packages/client/src/generation/utils.ts:138]: The `|| 'findOne'` is an always truthy statement, so it was invalid altogether. Result from linting

File: src/packages/get-platform/pnpm-lock.yaml (hunk 4 above)
Comment (timsuchanek) [reply src/packages/get-platform/pnpm-lock.yaml:1]: Yep will remove

**Pull request merged**